src/backend/distributed/commands/extension.c

```diff
@@ -579,11 +589,22 @@ MarkExistingObjectDependenciesDistributedIfSupported()
 	/* remove duplicates from object addresses list for efficiency */
 	List *uniqueObjectAddresses = GetUniqueDependenciesList(resultingObjectAddresses);
 
+	/*
+	 * We should sync the new dependencies during ALTER EXTENSION because
+	 * we cannot know whether the nodes has already been upgraded or not. If
+	 * the nodes are not upgraded at this point, we cannot sync the object. Also,
+	 * when the workers upgraded, they'd get the same objects anyway.
+	 */
+	bool prevDependencyCreationValue = EnableDependencyCreation;
+	SetLocalEnableDependencyCreation(false);
```

Comment:
why are we not just passing a boolean to MarkObjectDistributed?

(move the code into a new variant that takes a boolean, and make MarkObjectDistributed a wrapper that uses EnableDependencyCreation)

Reply:
I think we are going to do something like that here: https://github.com/citusdata/citus/pull/5684